lld/test/ELF/x86-64-gotpc-relax-too-far.s

```diff
@@ -47,6 +50,13 @@ SECTIONS {
   data 0x80200000 : { *(data) }
 }
 #--- lds2
+SECTIONS {
+  .text.foo 0x100000 : { *(.text.foo) }
```

Comment:
Do we need any checks to go along with the new sections?

Reply:
Using a linker script to achieve the goal is too complex.
Use two input sections instead? The second input contains relocations while the first is for padding.

Reply:
> Do we need any checks to go along with the new sections?

What do you mean by `new sections`? This linker script is a copy of the first one, but with the output section address adjusted.

> Use two input sections instead? The second input contains relocations while the first is for padding.

if I'm understanding your request, using two input sections (e.g.)
```
.section .text,"ax"
.space 4096

.section .text,"ax"
.globl _start
.type _start, @function
_start:
  movl __stop_data@GOTPCREL(%rip), %eax  # out of range
  movq __stop_data@GOTPCREL(%rip), %rax  # out of range
  movq __stop_data@GOTPCREL(%rip), %rax  # in range
  movl __stop_data@GOTPCREL(%rip), %eax  # in range
```
adjusts `rel.offset` instead of `sec->outSecOff` so it doesn't trigger this bug

the bug I found was in the context of a linker script

Reply:
> Do we need any checks to go along with the new sections?

if you mean `.text.foo`, perhaps we should remove that part of this test since it's obsoleted by the other part

Reply:
We've added new sections. Presumably it affects the output, but we're not checking for anything.
Basically, if we remove the added part, or if it has no intended effect, the test should start failing.

Reply:
The lds2 test does fail if `. += 0x1000` below is removed or `outSecOff` from code is removed. So the test seems fine to me.